### ISSUE TITLE
Lint/ParenthesesAsGroupedExpression: a false positive case with hash rocket argument

### ISSUE THREAD

**Issue by deepj** [rubocop-hq/rubocop#8834]
The following case uses `state_machine` DSL for describing an event of state machine inside a Rails model.

## Expected behavior

```ruby
transition (any - %i[failed canceled processing]) => :processing
```

The code doesn't violate `Lint/ParenthesesAsGroupedExpression` cop

## Actual behavior

```ruby
W: Lint/ParenthesesAsGroupedExpression: (...) interpreted as grouped expression.
      transition (any - %i[failed canceled processing]) => :processing
                ^
```

Auto-corrected to:

```ruby
transition(any - %i[failed canceled processing]) => :processing
```

This breaks code. Since the argument for `transition` method is `(any - %i[failed canceled processing]) => :processing`

## Steps to reproduce the problem

Code snippet
```ruby
    event :regenerate do
      transition (any - %i[failed canceled processing]) => :processing
    end
```

Run

```
rubocop --only Lint/ParenthesesAsGroupedExpression
```

## RuboCop version

```
$ rubocop -V
0.92.0 (using Parser 2.7.1.5, rubocop-ast 0.7.1, running on ruby 2.7.1 x86_64-darwin18)
```
